packages/trader/src/Modules/CFD/Containers/cfd-dashboard.jsx

```diff
@@ -39,139 +39,135 @@ const LoadTab = ({ children, is_loading, loading_component, ...props }) => {
-    };
-
-    async componentDidMount() {
-        this.updateActiveIndex(this.getIndexToSet());
-        this.openResetPassword();
-        this.props.onMount();
-    }
-
-    componentWillUnmount() {
-        this.props.onUnmount();
-    }
-
-    componentDidUpdate() {
-        this.updateActiveIndex();
-        this.props.checkShouldOpenAccount();
-
-        if (this.props.is_logged_in) {
+const CFDDashboard = props => {
+    const [is_demo_enabled, setIs_demo_enabled] = React.useState(false);
+    const [is_real_enabled, setIs_real_enabled] = React.useState(false);
+    const [active_index, setActiveIndex] = React.useState(0);
+    // const [is_account_needed_modal_open, setIs_account_needed_modal_open] = React.useState(false);
+    const [is_demo_tab] = React.useState(true);
+    // const [required_account, setRequired_account] = React.useState({});
+    const [is_notification_loaded, setIs_notification_loaded] = React.useState(false);
+    const [password_manager, setPassword_manager] = React.useState({
+        is_visible: false,
+        selected_login: '',
+        selected_account: '',
+        selected_account_type: '',
+        selected_account_group: '',
+        selected_server: '',
+    });
+
+    React.useEffect(() => {
+        updateActiveIndex(getIndexToSet());
+        openResetPassword();
+        props.onMount();
+    }, []);
+
+    // componentWillUnmount() {
+    //     this.props.onUnmount();
+    // }
+
+    React.useEffect(() => {
+        return () => {
+            props.onUnmount();
+        };
+    }, []);
+
+    React.useEffect(() => {
+        updateActiveIndex();
+        props.checkShouldOpenAccount();
+
+        if (props.is_logged_in) {
             ['demo', 'real'].forEach(account_type => {
                 const should_enable_tab =
-                    this.isSyntheticCardVisible(account_type) ||
-                    this.isFinancialCardVisible() ||
-                    this.isFinancialStpCardVisible();
+                    isSyntheticCardVisible(account_type) || isFinancialCardVisible() || isFinancialStpCardVisible();
 
-                const is_tab_enabled = this.state[`is_${account_type}_enabled`];
+                let is_tab_enabled = [`is_${account_type}_enabled`];
 
                 if (is_tab_enabled !== should_enable_tab) {
-                    this.setState({
-                        [`is_${account_type}_enabled`]: should_enable_tab,
-                    });
+                    is_tab_enabled = should_enable_tab;
                 }
             });
         }
-        const is_real_disabled = !this.state.is_real_enabled;
-        const is_demo_disabled = !this.state.is_demo_enabled;
-        if (!this.props.is_logged_in && (is_real_disabled || is_demo_disabled)) {
-            this.setState({
-                is_real_enabled: true,
-                is_demo_enabled: true,
-            });
+        const is_real_disabled = !is_real_enabled;
+        const is_demo_disabled = !is_demo_enabled;
```

Comment:
No need for these 2 lines as we only use them in one place! You can add them in line no.93